compiler/src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -1018,17 +1021,28 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
    */
   class MapToUnderlying extends TreeMap {
     override def transform(tree: Tree)(implicit ctx: Context): Tree = tree match {
-      case tree: Ident if tree.symbol.exists && !tree.symbol.owner.isClass && !tree.symbol.isAnonymousFunction && skipLocal(tree.symbol) =>
+      case tree: Ident if isBinding(tree.symbol) && skipLocal(tree.symbol) =>
```

Comment:
It seems type symbols should be skipped too? `isBinding` currently only ignores class symbols.